File: Java/marketplace/CreateAnticipation.java
package examples.marketplace;

import me.pagar.api.PagarmeCoreApiClient;
import me.pagar.api.controllers.RecipientsController;
import me.pagar.api.models.CreateAnticipationRequest;
import me.pagar.api.models.GetAnticipationResponse;
import me.pagar.api.http.client.*;
import org.joda.time.DateTime;

public class CreateAnticipation {

    public static void main(String[] args) {

        String basicAuthUserName = "ak_test_4tdVXpseumRmqbo"; // The username to use with basic authentication
        String basicAuthPassword = ""; // The password to use with basic authentication

        PagarmeCoreApiClient client = new PagarmeCoreApiClient(basicAuthUserName, basicAuthPassword);

        RecipientsController recipients_controller = new RecipientsController();

        CreateAnticipationRequest request = new CreateAnticipationRequest();

        request.setAmount(100);
        request.setTimeframe("start");
        request.setPaymentDate(DateTime.parse("2019-08-21"));

        String recipientId = "rp_RElaP4NMCJu08V9m";

        recipients_controller.createAnticipationAsync(recipientId, request, null,
                new APICallBack<GetAnticipationResponse>() {
                    @Override
                    public void onSuccess(HttpContext context, GetAnticipationResponse response) {
                        System.out.println("Anticipation create!");
                        System.out.println("Anticipation ID: " + response.getId());
                    }

                    @Override
                    public void onFailure(HttpContext context, Throwable error) {

                        System.out.println("Status response: " + context.getResponse().getStatusCode());
                        System.out.println(error.getMessage());
                        error.printStackTrace();

                    }
                });
    }
}

File: Python/marketplace/create_anticipation.py
from pagarmecoreapi.pagarmecoreapi_client import PagarmecoreapiClient
from pagarmecoreapi.models import *
from pagarmecoreapi.controllers import *
from pagarmecoreapi.exceptions.error_exception import *

# Configuration parameters and credentials
basic_auth_user_name = 'basic_auth_user_name' # The username to use with basic authentication
basic_auth_password = 'basic_auth_password' # The password to use with basic authentication

PagarmecoreapiClient(basic_auth_user_name, basic_auth_password)

recipients_controller = recipients_controller.RecipientsController()

request = create_anticipation_request.CreateAnticipationRequest()
request.amount = 10000
request.timeframe = "start"
request.payment_date = "2020-12-12"

recipientId = "rp_RElaP4NMCJu08V9m"

try:
    result = recipients_controller.create_anticipation(recipientId, request)
    assert result is not None
    assert result.id is not None
    print("Anticipation id: ", result.id)
except ErrorException as ex:
    print(ex.message)
    print("Errors: ", ex.errors)
except Exception as ex:
    raise ex